src/backend/joanie/core/models/accounts.py

```diff
@@ -112,28 +113,38 @@ class Meta:
                 condition=models.Q(is_main=True),
                 fields=["owner"],
                 name="unique_main_address_per_user",
-            )
+            ),
+            models.CheckConstraint(
+                check=(
+                    models.Q(is_reusable=False) & models.Q(is_main=False)
+                    | models.Q(is_reusable=True)
```

Comment:
Could this be simplified ?
```suggestion
                    models.Q(is_reusable=True) | models.Q(is_main=False)
```